ampersand-search-select.js

```diff
@@ -0,0 +1,241 @@
+;(function() {
+  var d3 = require('d3');
+  var _ = require('lodash');
+
+  var AmpersandState = require('ampersand-state');
+  var AmpersandView = require('ampersand-view');
+  var AmpersandSubCollection = require('ampersand-subcollection');
+
+  var SearchSelectState = AmpersandState.extend({
+    props: {
+      selected: 'object'
+    },
+    session: {
+      query: [ 'string', false, ''],
+      data: 'object',
+      idAttribute: 'string',
+      imageAttribute: 'string',
+      queryAttribute: 'string',
+
+      // Private Variables
+      _data: 'object',
+      _selected: [ 'array', false, function() { return []; } ],
+      _querySet: [ 'array', false, function() { return []; } ]
+    },
+    initialize: function() {
+      this._data = new AmpersandSubCollection(this.data);
+      this.selected = new AmpersandSubCollection(this.data);
+
+      this.on('change:_selected', function(selectState, selected) {
+        selectState.selected.configure({
+          filter: function(model) {
+            return !!~selected.indexOf(model[selectState.idAttribute]);
```

Comment:
Can you use `!== -1` instead of `!!~`?  Combining numbers, bitwise operators (involving sign flipping and two's complement) and boolean operators into a single expression is probably best avoided. Except maybe for code golf, but `n+1` beats `!!~n`, so this seems more like obfuscation :p